include/eve/module/core/regular/impl/simd/arm/sve/div.hpp

```diff
@@ -12,40 +12,66 @@
 
 namespace eve::detail
 {
-  template<conditional_expr C, arithmetic_scalar_value T, typename N, callable_options O>
-  EVE_FORCEINLINE wide<T,N> div_(EVE_REQUIRES(sve_),
-                                 C          const& cx,
-                                 O          const& o,
-                                 wide<T, N> const& v,
-                                 wide<T, N> const& w) noexcept
-  requires sve_abi<abi_t<T, N>>
+  template<callable_options O, arithmetic_scalar_value T, typename N>
+  EVE_FORCEINLINE wide<T,N> div_(EVE_REQUIRES(sve_), O const& opts, wide<T, N> a, wide<T, N> b) noexcept
+    requires sve_abi<abi_t<T, N>>
   {
-    if constexpr(O::contains(toward_zero) || O::contains(upward) ||
-                      O::contains(downward) || O::contains(to_nearest))
+    if constexpr (O::contains(saturated2))
+    {
+      return div.behavior(cpu_{}, opts, a, b);
+    }
+    else if constexpr(O::contains(toward_zero) || O::contains(upward) || O::contains(downward) || O::contains(to_nearest))
+    {
+      if (floating_value<T>) return round[opts](div(a, b));
+      else                   return div.behavior(cpu_{}, opts, a, b);
+    }
+    else
+    {
+      if constexpr (sizeof(T) >= 4)
+      {
+        return svdiv_x(sve_true<T>(), a, b);
+      }
+      else 
+      {
+        apply<N::value>([&](auto... I) { (a.set(I, a.get(I) / b.get(I)), ...); });
```

Comment:
this should be just in generic. Let's forward cpu.

Reply:
Then there is a bug that needs to be fixed.  Copy paste so that the bug doesn't trigger, is bad. Why is there an infinite.

Reply:
still an issue